ultra/setup.py

```diff
@@ -39,13 +35,13 @@
     description="Unprotected Left Turn for Robust Agents",
     long_description=long_description,
     long_description_content_type="text/markdown",
-    version="0.2.0",
+    version="0.2.1",
     packages=find_packages(exclude=["tests"]),
     include_package_data=True,
     zip_safe=True,
     python_requires=">=3.7",
     install_requires=[
-        "smarts[train,test]==0.4.14",
+        "smarts[train,test,camera-obs] @ git+https://git@github.com/huawei-noah/SMARTS@social-agents#egg=smarts",
```

Comment:
The smarts version specified on branch `social-agents` is v0.4.18, are we still going for upgrading smarts dep. to v0.4.19 for ULTRA?

Reply:
1. We will eventually update ULTRA to depend on the latest SMARTS release, namely v0.4.19, when it becomes available. 
2. Using SMARTS==0.4.18 causes `./tests/test_tune.py` to fail due to Ray not supporting `signal.signal`.
3. `signal.signal` occurs within the `RemoteAgentBuffer` class inside SMARTS and it is always instantiated by default upon SMARTS startup.
4. I modified SMARTS such that the `RemoteAgentBuffer` is only instantiated when social agents are needed. See #1037.
5. Since ULTRA does not use social agents currently, `RemoteAgentBuffer` would not be instantiated, and thus the `signal.signal` would not hinder our usage of Ray's `tune`.
6. In order to pass the tests and to keep up with the latest SMARTS optimizations, ULTRA's `setup.py` installs SMARTS from the branch `social-agents`. 
7. Currently the branch `social-agents` = [SMARTS==0.4.18 + some additional changes].
8. In summary, this is a handy method for ULTRA to use the latest SMARTS code, while waiting for the next release.

Reply:
Thank you @Adaickalavan for that information, it clearly make sense 👍 We might need to update ULTRA 's `setup.py` once SMARTS v0.4.19 (w/ `social-agents` branch updates) is released